### PR TITLE
[Merged by Bors] - Build go-spacemesh binaries for arm64

### LINKED ISSUE #4035: Provide arm64 linux and apple silicon builds for go-spacemesh
As in the title, we have self-hosted runners for that, so getting HW is not an issue.

### PR DESCRIPTION
**Pull request by brusherru**
## Motivation
It closes #4035 

## Changes
Added two more build targets and tweaked GHA workflow a bit.

## How to test
For testing purposes I've pushed tag [`v0.0.0-test-arm64`](https://github.com/spacemeshos/go-spacemesh/releases/tag/v0.0.0-test-arm64) — there are links to archives.
1. Download the archive corresponding to your os and arch
2. Unzip it
3. Go to extracted dir
4. Set chmod (on Linux & macOS): `chmod 755 ./*`
5. On MacOS turn off quarantine: `sudo xattr -rd com.apple.quarantine ./*`
6. Run go-spacemesh as usually (aka `./go-spacemesh -d ~/node-data -c ~/config.json`)
7. Ensure it has a full list of PoS providers. Call:
    ```
    grpcurl -d "{\"benchmark\": true}" -plaintext localhost:9092 spacemesh.v1.SmesherService.PostSetupComputeProviders
    ```
    The list depends on your OS, but hopefully, you'll see some GPU beside the CPU. :)

### CODE REVIEW

**Comment by brusherru** [comment]
bors try

File: version.txt [hunk 1]
```diff
@@ -1 +1 @@
-v0.0.0-unreleased
\ No newline at end of file
+v0.0.0-test-arm64
```
Comment (fasmat) [inline version.txt:1]: Is this change needed or can it be reverted?

**Review by fasmat — APPROVED** [review]

**Review by fasmat — COMMENTED** [review]
Generally looks fine to me, but I'm not sure if the two self hosted runners are intended for other things? @pigmej @andres-spacemesh

File: version.txt (hunk 1 above)
Comment (brusherru) [reply version.txt:1]: I believe it SHOULD be reverted :) I'll do it

Comment (brusherru) [reply version.txt:1]: Done ;)

**Comment by brusherru** [comment]
Reverted the latest commit which was made to test the workflow.
However, I keep the tag for a while, so you can still download artifacts and check them out.

I've tested both new macOS and Linux artifacts (inside VM). Both working well.

image

**Review by andres-spacemesh — APPROVED** [review]
I run the pipeline with https://github.com/nektos/act and dependencies looks ok on arm64

**Review by pigmej — APPROVED** [review]

**Review by pigmej — COMMENTED** [review]
We somehow include smrepl for darwin_amd64 in darwin_arm64 release.

I think we shall in separate pr remove smrepl at all from the release process.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -106,6 +121,8 @@ jobs:
             ## Zip Files
             - Windows: https://storage.googleapis.com/${{ secrets.GCP_BUCKET }}/${{ steps.version.outputs.content }}/Windows.zip
             - macOS: https://storage.googleapis.com/${{ secrets.GCP_BUCKET }}/${{ steps.version.outputs.content }}/macOS.zip
+            - macOS: https://storage.googleapis.com/${{ secrets.GCP_BUCKET }}/${{ steps.version.outputs.content }}/macOS_ARM64.zip
```
Comment (pigmej) [inline .github/workflows/release.yml]: we could add here that this is arm64

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -106,6 +121,8 @@ jobs:
...
             - Linux: https://storage.googleapis.com/${{ secrets.GCP_BUCKET }}/${{ steps.version.outputs.content }}/Linux.zip
+            - Linux: https://storage.googleapis.com/${{ secrets.GCP_BUCKET }}/${{ steps.version.outputs.content }}/Linux_ARM64.zip
```
Comment (pigmej) [inline .github/workflows/release.yml]: we could add here that this is arm64

**Comment by brusherru** [comment]
Ok, got rid of `smrepl` and tweak the release message a bit.
So I've pushed the "test" commit and tag one more time, after a successful build I'll revert it.

**Comment by brusherru** [comment]
bors merge
